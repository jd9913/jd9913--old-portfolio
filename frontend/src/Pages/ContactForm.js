import React, { useState } from "react";

const ContactForm = () => {
	const [email, setEmail] = useState("");
	const [message, setMessage] = useState("");
	const [name, setName] = useState("");

	const submitRequest = async (e) => {
		e.preventDefault();
		console.log({ name, email, message });
		const response = await fetch("/contact", {
			method: "POST",
			headers: {
				"Content-type": "application/json",
			},
			body: JSON.stringify({ name, email, message }),
		});
		const resData = await response.json();
		if (resData.status === "success") {
			alert("Message Sent.");
			this.resetForm();
		} else if (resData.status === "fail") {
			alert("Message failed to send.");
		}
	};

	return (
		<form onSubmit={submitRequest}>
			<div>
				<label htmlFor='name'>Name:</label>
				<input
					type='text'
					id='name'
					onChange={(e) => setName(e.target.value)}
					value={name}
					required
				/>
			</div>
			<div>
				<label htmlFor='email'>Email:</label>
				<input
					type='email'
					id='email'
					required
					onChange={(e) => setEmail(e.target.value)}
					value={email}
				/>
			</div>
			<div>
				<label htmlFor='message'>Message:</label>
				<textarea
					id='message'
					type='text'
					onChange={(e) => setMessage(e.target.value)}
					value={message}
					required
				/>
			</div>
			<button type='submit'>Submit</button>
		</form>
	);
};

export default ContactForm;
