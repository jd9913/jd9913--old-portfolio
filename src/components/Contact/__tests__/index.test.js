// __tests__/Contact.test.js
import React from 'react'
import { render, cleanup, getAllByTestId } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import ContactForm from '../../Contact';

afterEach(cleanup)

describe('Contact component renders', () => {
  it('renders', () => {
  render(<ContactForm />);


  const { asFragment } = render(<ContactForm />)
  expect(asFragment()).toMatchSnapshot()
})

})

describe('Contact component has content', ()=>{

it('renders',()=>{
  const { getByTestId } = render(<ContactForm />)
  expect(getByTestId('contactme')).toHaveTextContent('Contact me')
 
  expect(getByTestId('button')).toHaveTextContent('Submit')
  
})

})
  

