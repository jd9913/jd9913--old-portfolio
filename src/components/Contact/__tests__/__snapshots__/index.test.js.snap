// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component renders renders 1`] = `
<DocumentFragment>
  <section
    class="grid-container"
  >
    <h1
      data-testid="contactme"
    >
      Contact me
    </h1>
    <form
      id="contact-form"
    >
      <div>
        <label
          for="name"
        >
          Name:
        </label>
        <input
          name="name"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email address:
        </label>
        <input
          name="email"
          type="email"
          value=""
        />
      </div>
      <div>
        <label
          for="message"
        >
          Message:
        </label>
        <textarea
          name="message"
          rows="5"
        />
      </div>
      <button
        class="button"
        data-testid="button"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
